apis/eks/v1alpha1/types.go

```diff
@@ -0,0 +1,383 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// AMITypes is a type of AMI.
+type AMITypes string
```

Comment:
The same as below, however, in this case where this type is used is an input parameter. I think this is useful only if we get a validation. We are bitten by enum validation before as the set of values changed, in this case it doesn't seem to be that `AMITypes` will be changed but no one can say it cannot. So, I think we should either just go with `string` and have no validation  or use this type but also provide enum validation via kubebuilder tag.

Reply:
hmm yeah I don't know which to do here. I imagine that these will expand over time (potentially over a very long time), which makes me want to not restrict the allowed values. However, they are not necessarily intuitive values so that make me want to optimize for user experience. I think I will opt for just using `string` here since the values are denoted in the field description, and so folks will be able to look up allowed values on https://doc.crds.dev 👍